### DISCUSSION TITLE
Suggestion for next major release

### DISCUSSION THREAD

**Discussion by miyatsu** [immortalwrt/immortalwrt#1481] [General]
In last ImmortalWRT 23.05 release cycle, we have rc-1, rc-2, rc-3 and rc-4 tags applied prior to final release in order to follow up upstream tag timeline. This strategy put a lot of storage pressure to the repo server, I believe there is no one would using `rc-x` in production env. once we have final release tag applied, all of the *.ipk or *.bin with `rc-x` tag would be useless by then.

For next major release (possible OpenWRT 24.10), I recommand creating ONLY one release candidate tag (v24.10.0-rc1) for people who
want's to try the newest features. Wait until upstream release its final 24.10.0, ImmortalWRT tag 24.10 branch for the the second time with tag (v24.10.0).

This could save a lot of storage space on repo server, and having new branch release for testing ASAP as well.

Ping @1715173329

**Comment by 1715173329** [comment]
If you are maintaining a mirror, please note all -rc releases will be moved to [archive server](https://archive.immortalwrt.org) (with 301 redirection) at last and it will not take spaces downstream.

Actually we still have some users that use 23.05 -rc release nowadays.

**Comment by miyatsu** [comment]
Hi @1715173329 , I'm not a mirror maintainer but a normal user.

I've noticed that ImmortalWrt has postponed v23.05.0 release due to storage space concerns (see https://github.com/immortalwrt/immortalwrt/discussions/1095#discussioncomment-7382392). Which makes ImmortalWrt release date 1 month behind upstream OpenWrt.

As an normal ImmortalWrt user, I've waited another more month in last release cycle (till ImmortalWrt v23.05.0 got relased) before I'm upgrading my home's main router hardware (x86 box).

I'm hoping that I don't have to wait one more month (again) in next release cycle before I'm upgrading my router to ImmortalWrt v24.10.0. :(

**Reply by 1715173329** [reply]
So you just ignored the first half of the sentence?
There's nothing difference between OpenWrt 23.05.0-rc4 and OpenWrt 23.05.0. They are the same thing except version difference.

There's nothing "behind", or do you expect Debian and Ubuntu share the exactly same version?

**Reply by miyatsu** [reply]
AFAIK, an -rc release should never be used in production env, so I always kept away from using it by waiting the final release. Content was roughly the same between two versions, but users like me (not willing to using -rc releases) has waited one more month for v23.05.0 got released.

I'm not suggesting ImmortalWrt should stick to upstream release footprint, but we can keep the final version release date close to OpenWrt as much as possible.

**Reply by miyatsu** [reply]
If you have storage space concerns, maybe it's OK to tag ONLY twice on branch 24.10 (v24.10.0-rc1 and v24.10.0) I think?
